src/nvme/nbft.h

```diff
@@ -9,27 +9,1011 @@
+ * 		    If the @num_disc field is cleared to 0h, then this field
+ * 		    is reserved.
+ * @discv:	    Discovery Descriptor Version (DISCV): This field indicates
+ * 		    the version of the Discovery Descriptor.
+ * @num_disc:	    Number of Discovery Descriptors (NumDisc): This field indicates
+ * 		    the number of Discovery Descriptors (see &struct nbft_discovery),
+ * 		    if any, in the Discovery Descriptor List, if any.
+ * @reserved4:	    Reserved.
+ */
+struct __attribute__((__packed__)) nbft_control {
+	__u8 structure_id;
+	__u8 major_revision;
+	__u8 minor_revision;
+	__u8 reserved1;
+	__le16 csl;
+	__u8 flags;
+	__u8 reserved2;
+	struct nbft_heap_obj hdesc;
+	__u8 hsv;
+	__u8 reserved3;
+	__le32 hfio;
+	__le16 hfil;
+	__u8 hfiv;
+	__u8 num_hfi;
+	__le32 ssnso;
+	__le16 ssnsl;
+	__u8 ssnsv;
+	__u8 num_ssns;
+	__le32 seco;
+	__le16 secl;
+	__u8 secv;
+	__u8 num_sec;
+	__le32 disco;
+	__le16 discl;
+	__u8 discv;
+	__u8 num_disc;
+	__u8 reserved4[16];
+};
+
+/**
+ * enum nbft_control_flags - Control Descriptor Flags
+ * @NBFT_CONTROL_VALID:	Block Valid: indicates that the structure is valid.
+ */
+enum nbft_control_flags {
+	NBFT_CONTROL_VALID	= 1 << 0,
+};
+
+/**
+ * struct nbft_host - Host Descriptor (Figure 9)
+ * @structure_id: Structure ID: This field shall be set to 2h (i.e.,
+ * 		  Host Descriptor; #NBFT_DESC_HOST).
+ * @flags:	  Host Flags, see &enum nbft_host_flags.
+ * @host_id:	  Host ID: This field shall be set to the Host Identifier. This
+ * 		  field shall not be empty if the NBFT and NVMe Boot are supported
+ * 		  by the Platform.
+ * @host_nqn_obj: Host NQN Heap Object Reference: this field indicates a heap
+ * 		  object containing a Host NQN. This object shall not be empty
+ * 		  if the NBFT and NVMe Boot are supported by the Platform.
+ * @reserved:	  Reserved.
+ */
+struct __attribute__((__packed__)) nbft_host {
+	__u8 structure_id;
+	__u8 flags;
+	__u8 host_id[16];
+	struct nbft_heap_obj host_nqn_obj;
+	__u8 reserved[8];
+};
+
+/**
+ * enum nbft_host_flags - Host Flags
+ * @NBFT_HOST_VALID:			   Descriptor Valid: If set to 1h, then this
+ * 					   descriptor is valid. If cleared to 0h, then
+ * 					   this descriptor is reserved.
+ * @NBFT_HOST_HOSTID_CONFIGURED:	   HostID Configured: If set to 1h, then the
+ * 					   Host ID field contains an administratively-configured
+ * 					   value. If cleared to 0h, then the Host ID
+ * 					   field contains a driver default value.
+ * @NBFT_HOST_HOSTNQN_CONFIGURED:	   Host NQN Configured: If set to 1h, then the
+ * 					   Host NQN indicated by the Host NQN Heap Object
+ * 					   Reference field (&struct nbft_host.host_nqn)
+ * 					   contains an administratively-configured value.
+ * 					   If cleared to 0h, then the Host NQN indicated
+ * 					   by the Host NQN Offset field contains a driver
+ * 					   default value.
+ * @NBFT_HOST_PRIMARY_ADMIN_HOST_MASK:	   Mask to get Primary Administrative Host Descriptor:
```

Comment:
This definition is wrong.  Please change to: `NBFT_HOST_PRIMARY_ADMIN_MASK`

Reply:
Ah, docs, good catch! Fixed.

Reply:
I've tested this new updated branch and all of the documentation errors are fixed. The Fedora RPM now builds with not problem.